### DISCUSSION TITLE
How to use sequelize in Nuxt 3 server

### DISCUSSION THREAD

**Discussion by mazan-pavel** [nuxt/nuxt#15892] [Questions]
How to use sequelize(https://sequelize.org) for api  in Nuxt 3 server ?
 Is it possible to use this library for api requests? Or are there analogues for MariaDB?

If I use it as a plugin, it creates a lot of connectivity. It is enabled when navigating through pages.

this not work:
https://stackoverflow.com/questions/72859059/is-there-any-way-in-nuxt3-to-load-a-plugin-once

**Comment by Baroshem** [comment]
What about using a server plugin and an event handler from Nitro that you will be calling by using fetch? Plugin should be able to create a connection and event handler should be able to fetch the data from it?

Or use middleware as it used in sidebase to connect to Prisma? -> https://github.com/sidebase/sidebase/blob/main/server/middleware/0.prisma.ts

**Reply by Baroshem** [reply]
By looking at this example, you should be able to create a connection with middleware:

```ts


import { PrismaClient } from '@prisma/client'

let prisma: PrismaClient
declare module 'h3' {
  interface H3EventContext {
    prisma: PrismaClient
  }
}

export default eventHandler((event) => {
  if (!prisma) {
    prisma = new PrismaClient()
  }
  event.context.prisma = prisma
})
```

And then consume it from the api endpoint:

```ts
import type { H3Event } from 'h3'

export default eventHandler(async (event: H3Event) => {
  const prisma = event.context.prisma
  const { params } = event.context

  const example = await prisma.example.findFirst({ where: { id: params.id } })
  if (!example) {
    throw createError({ statusCode: 404, statusMessage: `Failed to find example with id ${params.id}` })
  }

  return example
})
```

So you should be able to do something similar with Sequelize.
